enarx-keep-sgx-shim/src/event.rs

```diff
@@ -32,32 +31,34 @@ pub extern "C" fn event(
         Some(Exception::InvalidOpcode) => {
             match unsafe { core::slice::from_raw_parts(h.aex.gpr.rip.raw() as *const u8, 2) } {
                 OP_SYSCALL => {
-                    aex.gpr.rax = Register::from_raw(match h.aex.gpr.rax.raw().into() {
-                        SysCall::READ => h.read(),
-                        SysCall::READV => h.readv(),
-                        SysCall::WRITE => h.write(),
-                        SysCall::WRITEV => h.writev(),
-                        SysCall::EXIT => h.exit(None),
-                        SysCall::GETUID => h.getuid(),
-                        SysCall::ARCH_PRCTL => h.arch_prctl(),
-                        SysCall::EXIT_GROUP => h.exit_group(None),
-                        SysCall::SET_TID_ADDRESS => h.set_tid_address(),
-                        SysCall::BRK => h.brk(),
-                        SysCall::UNAME => h.uname(),
-                        SysCall::MPROTECT => h.mprotect(),
-                        SysCall::MMAP => h.mmap(),
-                        SysCall::MUNMAP => h.munmap(),
-                        SysCall::RT_SIGACTION => h.rt_sigaction(),
-                        SysCall::RT_SIGPROCMASK => h.rt_sigprocmask(),
-                        SysCall::SIGALTSTACK => h.sigaltstack(),
+                    aex.gpr.rax =
+                        Register::from_raw(match h.aex.gpr.rax.raw().try_into().unwrap() {
```

Comment:
We can't `unwrap()` in the shim. It [causes an infinite loop when it fails](https://github.com/enarx/enarx/blob/master/enarx-keep-sgx-shim/src/main.rs#L17).